sdk/monitor/azure-monitor-ingestion/src/samples/java/com/azure/monitor/ingestion/UploadLogsAsyncClientSample.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.ingestion;
+
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.monitor.ingestion.models.UploadLogsResult;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sample to demonstrate uploading logs to Azure Monitor using the Async client.
+ */
+public class UploadLogsAsyncClientSample {
+
+    private static final Duration TIMEOUT = Duration.ofSeconds(10);
+    /**
+     * Main method to run the sample.
+     * @param args ignore args.
+     */
+    public static void main(String[] args) {
+        LogsIngestionAsyncClient client = new LogsIngestionClientBuilder()
+            .endpoint("<data-collection-endpoint>")
+            .credential(new DefaultAzureCredentialBuilder().build())
+            .buildAsyncClient();
+
+        List<Object> dataList = getLogs();
+        try {
+
+            // More details on Mono<> can be found in the project reactor documentation at :
+            // https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html
+
+            Mono<UploadLogsResult> resultMono = client.upload("<data-collection-rule-id>",
+                "<stream-name>", dataList);
+
+            UploadLogsResult result = resultMono.block(TIMEOUT);
```

Comment:
We should use `subscribe` instead of `block` to actually make this an async sample. We can then add a wait below the async sample with a comment to indicate that wait is required as the async call is non-blocking and the program terminates without it.

Reply:
I just wanted to keep it simple by using block. But yes we can make it a more generic async sample. I have changed it to using subscribe instead